modules/javafx.controls/src/main/java/javafx/scene/control/MultipleSelectionModelBase.java

```diff
@@ -864,8 +873,48 @@ public int bitsetSize() {
         }
 
         @Override public int indexOf(Object obj) {
-            reset();
-            return super.indexOf(obj);
+            if (!(obj instanceof Number)) {
+                return -1;
+            }
+            Number n = (Number) obj;
+            int index = n.intValue();
+            if(!bitset.get(index)) {
+                return -1;
+            }
+            // is left most bit
+            if(index==0) {
+                return 0;
+            }
+
+            // is right most bit
+            if( index == bitset.length()-1 ){
+                return size()-1;
+            }
+
+            // count right bit
+            if( index > bitset.length()/2 ) {
+                int count = 1;
+                for (;;) {
+                    index = bitset.nextSetBit(index + 1);
+                    if (index < 0) {
```

Comment:
Since it cannot be loaded with a smaller number of items than Integer.MAX_VALUE (it looks like it freezes), overflow does not occur in the actual usage environment.